src/wast-parser.y

```diff
@@ -650,9 +652,10 @@ expr1 :
       assert(if_->type == ExprType::If);
       if_->if_.true_->label = $2;
     }
-  | try_check LPAR BLOCK labeling_opt block RPAR catch_list {
-      $5->label = $4;
-      Expr* try_ = Expr::CreateTry($5, $7.first);
+  | try_check labeling_opt LPAR BLOCK labeling_opt block RPAR catch_list {
```

Comment:
Oh, I guess I missed this before. It seems to me that try is the block, and you shouldn't need both, right?
So it should be `try_check labeling_opt block catch_list` instead. Or does that not work?